opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -153,6 +153,12 @@ namespace Opm {
         const std::map<int, VFPInjTable>& getVFPInjTables() const;
         size_t getNumPhases() const;
 
+        // The saturation function family.
+        // If SWOF and SGOF are specified in the deck it return 1
+        // If SWFN, SGFN and SOF3 are specified in the deck it return 2
+        // If keywords are missing or mixed, an error is given.
+        const size_t getSaturationFunctionFamily() const;
```

Comment:
> That has been my "wet dream" for the saturation tables in the parser - but I have no clue how large a job that would be.

I also think it would be nice, but I also don't really see how this can be done because there may be 2D tables, etc. the only thing which I can come up with is to merge the fluid-matrix interactions into the parser. (IMO, that would throw out the baby with the bathwater, though.)